commercetools-internal-processors/src/main/java/io/sphere/sdk/annotations/processors/generators/BaseAbstractGenerator.java

```diff
@@ -0,0 +1,112 @@
+package io.sphere.sdk.annotations.processors.generators;
+
+import com.squareup.javapoet.FieldSpec;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.TypeName;
+import io.sphere.sdk.annotations.processors.models.PropertyGenModel;
+import io.sphere.sdk.annotations.processors.models.TypeUtils;
+
+import javax.annotation.Nullable;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.util.Elements;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+abstract class BaseAbstractGenerator {
+
+    protected final Elements elements;
+    protected final TypeUtils typeUtils;
+
+    BaseAbstractGenerator(final Elements elements) {
+        this.typeUtils = new TypeUtils(elements);
+        this.elements = elements;
+    }
+
+    protected FieldSpec createField(final PropertyGenModel property) {
+        return createField(property, Modifier.PRIVATE);
+    }
+
+    protected FieldSpec createField(final PropertyGenModel property, final Modifier modifier) {
+        final FieldSpec.Builder builder = createFieldBuilder(property, modifier);
+        return builder.build();
+    }
+
+    protected FieldSpec.Builder createFieldBuilder(final PropertyGenModel property, final Modifier modifier) {
+        final FieldSpec.Builder builder = FieldSpec.builder(property.getType(), property.getJavaIdentifier())
+                .addModifiers(modifier);
+
+        if (property.isOptional()) {
+            builder.addAnnotation(Nullable.class);
+        }
+        return builder;
+    }
+
+    /**
+     * Escapes the given name with an {@code "_"} if it's a java keyword (e.g. {@code default}.
+     *
+     * @param name the name to escape
+     * @return the escaped name
+     */
+    protected String escapeJavaKeyword(final String name) {
+        return SourceVersion.isKeyword(name) ? "_" + name : name;
+    }
+
+    /**
+     * Returns all property methods - including inherited methods - sorted by their property name.
+     *
+     * @param typeElement the type element
+     * @return methods sorted by their {@link PropertyGenModel#getPropertyName(ExecutableElement)}
+     */
+    protected List<ExecutableElement> getAllPropertyMethodsSorted(TypeElement typeElement) {
+        return typeUtils.getAllPropertyMethods(typeElement)
+                .sorted(Comparator.comparing(methodName -> escapeJavaKeyword(PropertyGenModel.getPropertyName(methodName))))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns all property methods - sorted by their property name.
```

Comment:
We just copied the javadoc from the original method, but we should update the javadoc because this method only returns the direct properties and not `all`.